google_guest_agent/non_windows_accounts.go

```diff
@@ -345,12 +345,12 @@ func createUserGroupCmd(cmd, user, group string) *exec.Cmd {
 // createGoogleUser creates a Google managed user account if needed and adds it
 // to the configured groups.
 func createGoogleUser(user string) error {
-	var uid string
+	var uid, gid string
 	if config.Section("Accounts").Key("reuse_homedir").MustBool(false) {
-		uid = getUID(fmt.Sprintf("/home/%s", user))
+		uid, gid = getUIDAndGID(fmt.Sprintf("/home/%s", user))
```

Comment:
Given that the `useradd_cmd` can be configured, the user may have changed it to pass in the `-d` argument and the home directory will be set to somewhere other than `/home`. We have a couple of options to avoid it happening. I'll not block this PR because of it since the behavior was introduced by another patch and got the behavior integrated already. Here's my ask: Please add a TODO comment around the path derivation in this function stating the problem I described.